packages/multichain-network-controller/src/MultichainNetworkController.ts

```diff
@@ -0,0 +1,195 @@
+  multichainNetworksMetadata: {},
+  nonEvmSelected: false,
+};
+
+/**
+ * Returns the state of the {@link MultichainNetworkController}.
+ */
+export type MultichainNetworkControllerGetStateAction =
+  ControllerGetStateAction<
+    typeof controllerName,
+    MultichainNetworkControllerState
+  >;
+
+/**
+ * Event emitted when the state of the {@link MultichainNetworkController} changes.
+ */
+export type MultichainNetworkStateControllerStateChange =
+  ControllerStateChangeEvent<
+    typeof controllerName,
+    MultichainNetworkControllerState
+  >;
+
+/**
+ * Actions exposed by the {@link MultichainNetworkController}.
+ */
+export type MultichainNetworkStateControllerActions =
+  MultichainNetworkControllerGetStateAction;
+
+/**
+ * Events emitted by {@link MultichainNetworkController}.
+ */
+export type MultichainNetworkControllerEvents =
+  MultichainNetworkStateControllerStateChange;
+
+/**
+ * Actions that this controller is allowed to call.
+ */
+export type AllowedActions =
+  | NetworkControllerGetStateAction
+  | NetworkControllerSetActiveNetworkAction
+  | AccountsControllerSetSelectedAccountAction
+  | NetworkControllerGetNetworkConfigurationByNetworkClientId;
+
+/**
+ * Events that this controller is allowed to subscribe.
+ */
+export type AllowedEvents = NetworkControllerStateChangeEvent;
+
+/**
+ * Messenger type for the MultichainNetworkController.
+ */
+export type MultichainNetworkControllerMessenger =
+  RestrictedControllerMessenger<
+    typeof controllerName,
+    MultichainNetworkStateControllerActions | AllowedActions,
+    MultichainNetworkControllerEvents | AllowedEvents,
+    AllowedActions['type'],
+    AllowedEvents['type']
+  >;
+
+/**
+ * {@link MultichainNetworkController}'s metadata.
+ *
+ * This allows us to choose if fields of the state should be persisted or not
+ * using the `persist` flag; and if they can be sent to Sentry or not, using
+ * the `anonymous` flag.
+ */
+const multichainNetworkControllerMetadata = {
+  multichainNetworkConfigurationsByChainId: { persist: true, anonymous: false },
+  selectedMultichainNetworkChainId: { persist: true, anonymous: false },
+  multichainNetworksMetadata: { persist: true, anonymous: false },
+  nonEvmSelected: { persist: true, anonymous: false },
+};
+
+/**
+ * The MultichainNetworkController is responsible for fetching and caching account
+ * balances.
+ */
+export class MultichainNetworkController extends BaseController<
+  typeof controllerName,
+  MultichainNetworkControllerState,
+  MultichainNetworkControllerMessenger
+> {
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: MultichainNetworkControllerMessenger;
+    state: MultichainNetworkControllerState;
+  }) {
+    super({
+      messenger,
+      name: controllerName,
+      metadata: multichainNetworkControllerMetadata,
+      state: {
+        ...defaultState,
+        ...state,
+      },
+    });
+  }
+
+  async setActiveNetwork(clientId: string, chainId?: string): Promise<void> {
```

Comment:
Should also be a `CaipChainId` rather than a `string` here.

I include a suggestion, but you will need to add the associated `import { CaipChainId } from '@metamask/utils'` yourself 😅

```suggestion
  async setActiveNetwork(clientId: string, chainId?: CaipChainId): Promise<void> {
```